### PR TITLE
Enhance the search so that marker stays on feature

### PR DESCRIPTION
**Pull request by luciman**
Fixes https://github.com/qgis/qwc2-demo-app/issues/23

### CODE REVIEW

File: QWC2Components/components/Search.jsx [hunk 1]
```diff
@@ -336,8 +335,44 @@ const Search = React.createClass({
         if(item === this.state.currentResult) {
             let feature = IdentifyUtils.wktToGeoJSON(geometry);
             feature.geometry = IdentifyUtils.reprojectFeatureGeometry(feature.geometry, crs, this.props.mapConfig.projection);
+            let geojson  = new ol.format.GeoJSON().readFeature(feature)
+            let center = this.getFeatureCenter(geojson.getGeometry())
+            let wgscenterlatlon = CoordinatesUtils.reproject(center, item.crs, "EPSG:4326");
+            let text = item.label !== undefined ? item.label : item.text;
```
Comment (manisandro) [inline QWC2Components/components/Search.jsx]: Isn't this like a duplicate with line 297?

**Comment by luciman** [comment]
```let text = item.label !== undefined ? item.label : item.text;``` is duplicate.
Indeed. The problem is that I'm a reactjs newbie and I don't know how to pass the text variable to showFeatureGeometry() I'll refactor this once I learn how to do it. Any hints are appreciated.
Thanks.

**Comment by manisandro** [comment]
You can just use an inline function 

    (item, geometry, crs) => { showFeatureGeometry(item, geometry, crs, label) }

but in this case, just remove the first `let text [...]` line, it's unused AFAICS.

**Comment by luciman** [comment]
@manisandro Thanks for the advice. It's reworked now. I also had an error for the marker on coordinate search.

**Comment by manisandro** [comment]
Thanks!

**Pull request merged**